apps/admin-panel/app/api/auth/[...nextauth]/options.ts

```diff
@@ -57,6 +79,41 @@ const callbacks: Partial<CallbacksOptions> = {
     const verified = new Boolean("email_verified" in profile && profile.email_verified)
     return verified && env.AUTHORIZED_EMAILS.includes(email)
   },
+  async jwt({ token, user }) {
+    let role_mapping: { [key: string]: string }
+    if (env.NODE_ENV === "development") {
+      role_mapping = {
+        "admintest@blinkbitcoin.test": "ADMIN",
+        "alicetest@blinkbitcoin.test": "VIEWER",
+        "bobtest@blinkbitcoin.test": "SUPPORT",
+      }
+    } else {
+      role_mapping = env.USER_ROLE_MAP
+    }
+    if (user) {
+      const userRole = role_mapping[user.email as keyof typeof role_mapping] || "VIEWER"
+
+      // Validate and get access rights for the role
+      if (isValidAdminRole(userRole)) {
+        const accessRights = getAccessRightsForRole(userRole as AdminRole)
+        token.scope = JSON.stringify(accessRights)
+        token.role = userRole
```

Comment:
Should we add the role? The api is not using it but it might be good for the refernce/debugging.